src/model_loader.cpp

```diff
@@ -0,0 +1,111 @@
+#include "model_loader.hpp"
+
+namespace wr
+{
+	ModelLoader::~ModelLoader()
+	{
+		for (ModelData* model : m_loaded_models)
```

Comment:
Consider using the DeleteModel function instead of duplicating code. There are ways to safely loop through a vector while deleting elements in it, but if that requires too much time and you have other tasks I can still accept this.